agent/util-scripts/unittests

```diff
@@ -136,17 +138,21 @@ function _run {
 function _save_tree {
     # Save state of the tree
     echo "+++ pbench tree state" >> ${_testout}
-    find ${_testdir} 2> /dev/null | sort >> ${_testout}
-
-    for x in $(find "${_testdir}/tools"-* -type f 2> /dev/null | sort); do
-        echo "=== ${x}:"
-        cat ${x}
-    done >> ${_testout}
-
-    for x in $(find "${_testdir}/tmp" -type f 2> /dev/null | sort); do
-        echo "=== ${x}:"
+    find ${_testdir} 2> /dev/null \
+		| sed -e 's;tmp/\([a-z-]*\)\.[0-9][0-9]*;tmp/\1.NNNNN;g' \
+		  	  -e 's;\(tm\.[a-z-]*\)\.[0-9][0-9]*\.[a-z0-9_][a-z0-9_]*;\1.NNNNN.nnnnnnnn;g' \
+        | sort >> ${_testout}
+
+    for x in $(find "${_testdir}/tools"-* "${_testdir}/tmp" -type f 2> /dev/null | sort); do
+		y=$(echo "${x}" | sed -e 's;tmp/\([a-z-]*\)\.[0-9][0-9]*;tmp/\1.NNNNN;g' \
```

Comment:
This looks like it's indented a stop further than it should be? It's hard enough to read this massive `sed` munging!

So where is this `tmp/<blah>.NNNNN` or `tm.<blah>.NNNNN.nnnnnnnnn` pattern triggering? I looked through some of the `util-scripts` gold files and don't see an obvious sign of that?

Reply:
> This looks like it's indented a stop further than it should be?

Agreed, this is a tabs vs spaces issues, it appears.  I'll fix.

> It's hard enough to read this massive `sed` munging!

Agreed.  Ah, "gold file" based unit tests...

> So where is this `tmp/<blah>.NNNNN` or `tm.<blah>.NNNNN.nnnnnnnnn` pattern triggering? I looked through some of the `util-scripts` gold files and don't see an obvious sign of that?

Well I'll take it out when I fix this problem and report back.

Reply:
For the 1st sed, `s;tmp/\([a-z-]*\)\.[0-9][0-9]*;tmp/\1.NNNNN;g`:

```
*** /home/pportant/Documents/pbench/agent/util-scripts/gold/pbench-copy-results/test-31.txt     2021-02-10 23:05:26.495795365 -0500
--- /var/tmp/pbench-test-utils/output.txt       2021-02-17 12:52:44.463849005 -0500
***************
*** 1,5 ****
  +++ Running test-31 pbench-copy-results --prefix=foo/bar --user=ndk
! tar --create --force-local "pbench-user-benchmark_ndk-test-1_2018.05.23T03.21.32" | xz -T0 > "/var/tmp/pbench-test-utils/pbench/tmp/pbench-copy-results.NNNNN/test>
  --- Finished test-31 pbench-copy-results (status=0)
  +++ pbench tree state
  /var/tmp/pbench-test-utils/pbench
--- 1,5 ----
  +++ Running test-31 pbench-copy-results --prefix=foo/bar --user=ndk
! tar --create --force-local "pbench-user-benchmark_ndk-test-1_2018.05.23T03.21.32" | xz -T0 > "/var/tmp/pbench-test-utils/pbench/tmp/pbench-copy-results.3671762/te>
  --- Finished test-31 pbench-copy-results (status=0)
  +++ pbench tree state
  /var/tmp/pbench-test-utils/pbench
```

For the 2nd sed, `s;\(tm\.[a-z-]*\)\.[0-9][0-9]*\.[a-z0-9_][a-z0-9_]*;\1.NNNNN.nnnnnnnn;g`:

```
*** /home/pportant/Documents/pbench/agent/util-scripts/gold/test-client-tool-meister/test-56.txt        2021-02-17 12:49:37.918076890 -0500
--- /var/tmp/pbench-test-utils/output.txt       2021-02-17 12:55:14.188468715 -0500
***************
*** 153,162 ****
  /var/tmp/pbench-test-utils/pbench/mock-run/tools-lite/prometheus_data.tar.gz
  /var/tmp/pbench-test-utils/pbench/pbench.log
  /var/tmp/pbench-test-utils/pbench/tmp
! /var/tmp/pbench-test-utils/pbench/tmp/tm.lite.NNNNN.nnnnnnnn
! /var/tmp/pbench-test-utils/pbench/tmp/tm.lite.NNNNN.nnnnnnnn/red:remote_c.example.com
! /var/tmp/pbench-test-utils/pbench/tmp/tm.lite.NNNNN.nnnnnnnn
! /var/tmp/pbench-test-utils/pbench/tmp/tm.lite.NNNNN.nnnnnnnn/red:remote_c.example.com
  /var/tmp/pbench-test-utils/pbench/tools-v1-lite
  /var/tmp/pbench-test-utils/pbench/tools-v1-lite/remote_a.example.com
  /var/tmp/pbench-test-utils/pbench/tools-v1-lite/remote_a.example.com/mpstat
--- 153,162 ----
  /var/tmp/pbench-test-utils/pbench/mock-run/tools-lite/prometheus_data.tar.gz
  /var/tmp/pbench-test-utils/pbench/pbench.log
  /var/tmp/pbench-test-utils/pbench/tmp
! /var/tmp/pbench-test-utils/pbench/tmp/tm.lite.3681729.ila650u1
! /var/tmp/pbench-test-utils/pbench/tmp/tm.lite.3681729.ila650u1/red:remote_c.example.com
! /var/tmp/pbench-test-utils/pbench/tmp/tm.lite.3681729.wh42brk0
! /var/tmp/pbench-test-utils/pbench/tmp/tm.lite.3681729.wh42brk0/red:remote_c.example.com
  /var/tmp/pbench-test-utils/pbench/tools-v1-lite
  /var/tmp/pbench-test-utils/pbench/tools-v1-lite/remote_a.example.com
  /var/tmp/pbench-test-utils/pbench/tools-v1-lite/remote_a.example.com/mpstat
```